compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -1821,14 +1821,24 @@ error.invalid.field.access.in.record.type=\
   may have been specified as a rest field
 
 error.undeclared.and.optional.fields.in.union.of.records=\
-  field access can only be used to access required fields, field ''{0}'' is undeclared in record(s) ''{1}'' and \
-  optional in record(s) ''{2}''
+  field access can only be used to access required fields or optional fields of non-nilable types, field ''{0}'' \
+  is undeclared in record(s) ''{1}'' and optional in record(s) ''{2}''
 
 error.undeclared.field.in.union.of.records=\
-  field access can only be used to access required fields, field ''{0}'' is undeclared in record(s) ''{1}''
+  field access can only be used to access required fields or optional fields of non-nilable types, field ''{0}'' \
+  is undeclared in record(s) ''{1}''
 
 error.optional.field.in.union.of.records=\
-  field access can only be used to access required fields, field ''{0}'' is optional in record(s) ''{1}''
+  field access can only be used to access required fields or optional fields of non-nilable types, field ''{0}'' \
+  is optional in record(s) ''{1}''
+
+error.undeclared.and.optional.and.nilable.fields.in.union.of.records=\
+  field access can only be used to access required fields or optional fields of non-nilable types, field ''{0}'' \
+  is undeclared in record(s) ''{1}'', optional in record(s) ''{2}'' and type includes nil in record(s) ''{3}''
```

Comment:
Aren't ``''{2}'' and ''{3}''` effectively the same? We only need to mention the records in which the field is optional and the type is nilable?